src/libmongoc/src/mongoc/mongoc-topology.c

```diff
@@ -1817,3 +1824,23 @@ _topology_collect_errors (mongoc_topology_t *topology, bson_error_t *error_out)
                  sizeof (error_out->message));
    bson_string_free (error_message, true);
 }
+
+/* Caller must lock topology->mutex to protect handshake_cmd. */
+void
+_mongoc_topology_set_server_api (mongoc_topology_t *topology,
+                                 const mongoc_server_api_t *api)
+{
+   mongoc_server_description_t *server_description;
+   int i;
+
+   BSON_ASSERT (topology);
+   BSON_ASSERT (api);
+
+   /* Set hello_ok for all server descriptions */
```

Comment:
Is this necessary? The topology scanner returns the monitoring and handshake commands in `_mongoc_topology_scanner_get_handshake_cmd` and `_mongoc_topology_scanner_get_monitoring_cmd` as `hello` if a server API version is set. Those are used in the pooled server monitor as well.

Reply:
Good point. I added this at the same time as I added the checks for `api` when deciding whether to use `hello` or legacy hello. I've removed this function (as it would only proxy the call to the topology scanner), which also takes care of the unconditional declaration that I wasn't able to see 🤦‍♂️